src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -236,6 +235,33 @@ int ComputeTsBox(float redshift, float prev_redshift, struct UserParams *user_pa
             Use_Hawking_Radiation = false;
         }
 
+        // Re-setting, the FlagOptions overrides them all
+        // A question: should I complain (raise warning and proceed with FlagOptions) or abort (raise error) when numerical inputs are inconsistent with FlagOptions?
+        if ((!flag_options->USE_Radio_ACG) && (Use_Radio_ACG))
+        {
+            Use_Radio_ACG=false;
+            LOG_ERROR("Mission aborted due to conflicting params: you have fR>0 but FlagOptions->USE_Radio_ACG=F, you need to set FlagOptions->USE_Radio_ACG=T to use Radio ACG.");
+            Throw(ValueError);
+        }
+        if ((!flag_options->USE_Radio_MCG) && (Use_Radio_MCG))
+        {
+            Use_Radio_MCG=false;
+            LOG_ERROR("Mission aborted due to conflicting params: you have fR_mini>0 but FlagOptions->USE_Radio_MCG=F, you need to set FlagOptions->USE_Radio_MCG=T to use Radio MCG.");
+            Throw(ValueError);
+        }
+        if ((!flag_options->USE_Radio_PBH) && (Use_Radio_PBH))
+        {
+            Use_Radio_PBH=false;
+            LOG_ERROR("Mission aborted due to conflicting params: numerical params says Use_Radio_PBH but FlagOptions->USE_Radio_PBH=F, you need to set FlagOptions->USE_Radio_PBH=T to use Radio PBH.");
+            Throw(ValueError);
+        }
+        if ((!flag_options->USE_Hawking_Radiation) && (Use_Hawking_Radiation))
+        {
+            Use_Hawking_Radiation=false;
+            LOG_ERROR("Mission aborted due to conflicting params: numerical params says Use_Hawking_Radiation but FlagOptions->Use_Hawking_Radiation=F, you need to set FlagOptions->Use_Hawking_Radiation=T to use Hawking Radiation.");
+            Throw(ValueError);
+        }
+
```

Comment:
We actually deal with this kind of logic on the Python side, where it is a little bit easier to do so. Use the `py21cmfast.inputs.validate_all_inputs()` function to do these checks.

Also, you should not have both `flag_options->USE_Hawking_Radiation` AND `Use_Hawking_Radiation` defined. Just have one source of truth (the flag options).

Reply:
Yeah in the earlier versions the Use_Hawking_Radiation (and other bool variables) were used to determine whether certain new features should be turned on, but with the new FlagOptions these old variables seem unnecessary. In the new version I have removed these internal switches in SP.c.

Reply:
Inputs parameter checks are now moved from SP.c to inputs.py